types/block.go

```diff
@@ -56,6 +56,20 @@ func CalculateCoinbase(height BlockHeight) Currency {
 	return NewCurrency64(base).Mul(SiacoinPrecision)
 }
 
+// CalculateNumSiacoins calculates the number of siacoins in circulation at a
+// given height.
+func CalculateNumSiacoins(height BlockHeight) Currency {
+	deflationBlocks := BlockHeight(InitialCoinbase - MinimumCoinbase)
+	avgDeflationSiacoins := CalculateCoinbase(0).Add(CalculateCoinbase(height)).Div(NewCurrency64(2))
+	if height <= deflationBlocks {
+		deflationSiacoins := avgDeflationSiacoins.Mul(NewCurrency64(uint64(height + 1)))
+		return deflationSiacoins
+	}
+	deflationSiacoins := avgDeflationSiacoins.Mul(NewCurrency64(uint64(deflationBlocks + 1)))
+	trailingSiacoins := NewCurrency64(uint64(height - deflationBlocks)).Mul(CalculateCoinbase(height))
+	return deflationSiacoins.Add(trailingSiacoins)
```

Comment:
1 million bignum operations takes half a second. It doesn't hurt having an optimized version, especially since the test verifies its correctness, but readability suffers for no perceivable gain.